index.mjs

```diff
@@ -0,0 +1,181 @@
+import { env } from 'process';
+import { join } from 'path';
+import { readFileSync } from 'fs';
+import { execSync } from 'child_process';
+import crypto from 'crypto';
+import core from '@actions/core';
+import { context, getOctokit } from '@actions/github';
+
+const { GITHUB_WORKSPACE } = env;
+
+const commentIdentifier = '<!-- NPM_PUBLISH_BRANCH_COMMENT_PR -->';
+
+// returns an Octokit client that we use to make PR comments
+const getGithubClient = () => {
+  const githubToken = core.getInput('github_token');
+
+  if (!githubToken) {
+    throw new Error('No GitHub token provided');
+  }
+
+  if (!context.issue.number) {
+    throw new Error('This is not a PR or commenting is disabled.');
+  }
+
+  const client = getOctokit(githubToken);
+
+  if (!client) {
+    throw new Error('Client could not be created; ensure sure that GitHub token is correct.');
+  }
+
+  return client;
+};
+
+// iterate through comment list to find one that begins with the
+// hidden identifier we added in getPullRequestComment()
+const findComment = async (client) => {
+  const options = {
+    owner: context.issue.owner,
+    repo: context.issue.repo,
+    issue_number: context.issue.number,
+  };
+  const comments = await client.rest.issues.listComments(options);
+
+  let commentId = null;
```

Comment:
🙃  This could be a little more concise as well as short-circuited by using `.find` :)

```
  const comment = comments.data.find(({ body }) => body.startsWith(commentIdentifier))
  
  return comment !== undefined ? comment.id : null;
```

Reply:
@AustinBurns good call!! I like that better, [updated here](https://github.com/wistia/publish-branch-to-npm/pull/2/commits/2b02325a1962e5bc9377df558d21c4a3e5d26208#diff-2bf70ead4f52d013ec9fd4739e0f1e0a7c232987f48a75e464218170cf13b6a1R8449).